### ISSUE TITLE
EndDraw, GetSize, and GetPixelSize methods on ID2D1RenderTarget should return HRESULT

### ISSUE THREAD

**Issue by Shkyrockett** [microsoft/win32metadata#1314]

**Comment by kennykerr** [comment]
These methods all work correctly in Rust, so I can confirm the metadata is correct. Note that various Direct2D methods don't return `HRESULT` values. The `GetSize` and `GetPixelSize` methods return structs directly (depending on calling convention), which requires special handling for historical reasons. This was previously discussed here: https://github.com/microsoft/win32metadata/issues/636

**Comment by AArnott** [comment]
> These methods all work correctly in Rust, so I can confirm the metadata is correct.

I don't think that necessarily follows. Our projections interpret the metadata differently, so it's possible one works while a bug remains in the metadata. But I'm just exploring here and trying to understand. 

Thanks for the link to the other issue. I'd personally say the metadata is incorrect, since we can't take the declaration as given, and there is no annotation in the metadata to tell us that projections need to do something special here. It's just that Rust has already figured out this 'bug' in the metadata and worked around it, and CsWin32 hasn't.
It's particularly ironic because the metadata is literally telling us to _preserve_ the signature, when in fact the signature won't do the right thing.

`EndDraw` I see isn't defective... it's just a request like #1295 because it returns a failing HRESULT in success paths so avoiding the exception would be nice. So this is really about the `GetSize` and `GetPixelSize` methods.

**Comment by kennykerr** [comment]
Sure, you could argue the metadata could be improved. I'm just pointing out that it works today so we shouldn't blindly "fix" these methods. Another case of this: https://github.com/microsoft/win32metadata/issues/1295#issuecomment-1276606660

**Comment by AArnott** [comment]
FYI we'll track `EndDraw` as part of #1315.

**Comment by Shkyrockett** [comment]
I suggested maybe these need to be marked as `[PreserveSig]`, because if I am understanding the code and comments correctly for CsWin32; it looks like CsWin32 does some rewriting of methods in interfaces if the method is not marked as `[PreserveSig]`, and only has out, or zero parameters: https://github.com/microsoft/CsWin32/blob/main/src/Microsoft.Windows.CsWin32/Generator.cs#L3653

These would fall into those specifications.

**Comment by AArnott** [comment]
I'll move this back to CsWin32 and track fixing our generator.
